src/plugins/vis_augmenter/server/plugin.ts

```diff
@@ -23,42 +26,53 @@ export interface VisAugmenterPluginStart {}
 export class VisAugmenterPlugin
   implements Plugin<VisAugmenterPluginSetup, VisAugmenterPluginStart> {
   private readonly logger: Logger;
+  private readonly config$: Observable<VisAugmenterPluginConfigType>;
 
   constructor(initializerContext: PluginInitializerContext) {
     this.logger = initializerContext.logger.get();
+    this.config$ = initializerContext.config.create<VisAugmenterPluginConfigType>();
   }
 
-  public setup(core: CoreSetup) {
+  public async setup(core: CoreSetup) {
     this.logger.debug('VisAugmenter: Setup');
     core.savedObjects.registerType(augmentVisSavedObjectType);
     core.capabilities.registerProvider(capabilitiesProvider);
 
-    core.uiSettings.register({
-      ['visualization:enablePluginAugmentation']: {
-        name: i18n.translate('visualization.enablePluginAugmentationTitle', {
-          defaultMessage: 'Enable plugin augmentation',
-        }),
-        value: true,
-        description: i18n.translate('visualization.enablePluginAugmentationText', {
-          defaultMessage: 'Plugin functionality can be accessed from line chart visualizations',
-        }),
-        category: ['visualization'],
-        schema: schema.boolean(),
-      },
-      ['visualization:enablePluginAugmentation.maxPluginObjects']: {
-        name: i18n.translate('visualization.enablePluginAugmentation.maxPluginObjectsTitle', {
-          defaultMessage: 'Max number of associated augmentations',
-        }),
-        value: 10,
-        description: i18n.translate('visualization.enablePluginAugmentation.maxPluginObjectsText', {
-          defaultMessage:
-            'Associating more than 10 plugin resources per visualization can lead to performance ' +
-            'issues and increase the cost of running clusters.',
-        }),
-        category: ['visualization'],
-        schema: schema.number({ min: 0 }),
-      },
-    });
+    const config: VisAugmenterPluginConfigType = await this.config$.pipe(first()).toPromise();
+    const isAugmentationEnabled =
+      config.pluginAugmentationEnabled === undefined ? true : config.pluginAugmentationEnabled;
+
+    if (isAugmentationEnabled) {
```

Comment:
2 things:
1. is `isAugmentationEnabled` pulling the value from the yml file, and then in the below `if` block, registering the relevant `uiSettings` accordingly? And if so, adding a couple comments here could be helpful, since i think this is kind of a unique scenario where we need to set UI Settings based on a yml setting
2. Are we sure the default should be `true` if the setting is undefined?

Reply:
By default, the advanced setting is true. This is so when they first interact with the cluster, they can use this feature and they can disabled it if needed. The only time the setting is undefined is if `isAugmentationEnabled`, so we do not set this setting.

Reply:
Can you point me to where this is defined in the yml file and/or how the ui settings may be set based on this setting? And if not defined in the yml file, are we sure the default should be `true`?

Reply:
The settings is defined [here](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3961/files#diff-5d54489df9aac06050baae247879eddfb5057500423a631187245e7331238e0aR245).

By default, all the settings in the yaml file are to be commented out. If they want to disable it, they would set it to false there. When it is set to false, it doesn't register the UI settings, so when getting those settings, it would return as undefined. 

We want to enable this feature by default, so we want the default to be true.